src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -1545,6 +1545,39 @@ public static IEnumerable<IEdmOperationImport> OperationImports(this IEdmEntityC
             return container.AllElements().OfType<IEdmOperationImport>();
         }
 
+        /// <summary>
+        /// Returns NavigationPropertyBindings with their corresponding container elements (EntitySet or Singelton)
+        /// </summary>
+        /// <param name="container">Reference to the calling object.</param>
+        /// <returns>collection of pairs of container element and NavigationPropertyBindings.</returns>
+        public static IEnumerable<(IEdmEntityContainerElement, IEdmNavigationPropertyBinding)> GetNavigationPropertyBindings(this IEdmEntityContainer container)
+        {
+            EdmUtil.CheckArgumentNull(container, "container");
+
+            var elements = container.AllElements();
+            foreach(var element in elements)
+            {
+                switch (element)
+                {
+                    case IEdmEntitySet entitySet:
+                        foreach(var binding in entitySet.NavigationPropertyBindings)
+                        {
+                            yield return (entitySet, binding);
+                        }
+                        break;
+                    case IEdmSingleton singleton:
+                        foreach (var binding in singleton.NavigationPropertyBindings)

```

Comment:
It's a more fundamental change, but I wonder if it would make sense for `IEdmEntitySet` and `IEdmSingleton` to derive from a common interface that has `NavigationPropertyBindings` on it. Then this whole method is sort of obsolete (or at least very derivative):
```
return container.AllElements().OfType<IEdmWithNavPropertyBindings>().SelectMany(element => element.NavigationPropertyBindings.Select(binding => (element, binding)));
```